backend/app/crud/openai_conversation.py

```
@@ -0,0 +1,168 @@
+        OpenAIConversation.ancestor_response_id == ancestor_response_id,
+        OpenAIConversation.project_id == project_id,
+        OpenAIConversation.is_deleted == False,
+    )
+    result = session.exec(statement).first()
+    return result
+
+
+def set_ancestor_response_id(
+    session: Session,
+    current_response_id: str,
+    previous_response_id: Optional[str],
```

Ruff (0.12.2)

Use `X | None` for type annotations

Convert to `X | None`

(UP045)